### DISCUSSION TITLE
deunhealth - Able to restart the whole stack in Portainer?

### DISCUSSION THREAD

**Discussion by derrick087** [qdm12/deunhealth#51] [General]
Hi, I have a Portainer stack that is running a VPN with transmission plus other containers and after a while, the VPN will shut off every week going offline and the only way I found out to fix it is to stop and restart the stack within Portainer. I'm assuming it has something to do with the kill switch of the stack I'm running, it's when ever the container "alpine" goes offline the whole stack goes bad forcing me to stop and start the whole stack again to just to get it working again.

Is deunhealth able to restart the whole stack once any container goes unhealthy or offline in Portainer?

I'm still new to the docker containers scene and just put together what I learned from random videos. So if there is any other solution or something I'm not aware of I've attached a copy of my stack. Any help or comments on improvements are appreciated.

[my stack copy.txt](https://github.com/qdm12/deunhealth/files/9239685/my.stack.copy.txt)

**Comment by madcowGit** [comment]
for the containers that depend on vpn is use:

```
  depends_on:
    openvpn:
      condition: service_healthy    
```
this way the container exits when the vpn container is unhealthy. You can add a restart: always to have it restart again automatically. And since it depends on the vpn container it will wait for a healthy container untill it starts

**Comment by MatlouMacdonaldSekgaphola** [comment]
So if I want to archive this deunhealthy container using run insteat of compose, what should I do to update .env file?
